frontend/src/components/MeetingRoom/AudioInputOutputDevices/AudioInputOutputDevice.spec.tsx

```diff
@@ -1,9 +1,39 @@
-import { render, screen } from '@testing-library/react';
+import { act, queryByText, render, screen, waitFor } from '@testing-library/react';
 import { describe, beforeEach, it, Mock, vi, expect, afterAll } from 'vitest';
 import { MutableRefObject } from 'react';
+import { EventEmitter } from 'stream';
 import * as util from '../../../utils/util';
 import AudioInputOutputDevices from './AudioInputOutputDevices';
 import { AudioOutputProvider } from '../../../Context/AudioOutputProvider';
+import {
+  audioInputDevices,
+  audioOutputDevices,
+  nativeDevices,
+  videoInputDevices,
+} from '../../../utils/mockData/device';
+
+const {
+  mockHasMediaProcessorSupport,
+  mockGetDevices,
+  mockGetAudioOutputDevices,
+  mockSetAudioOutputDevice,
+  mockGetActiveAudioOutputDevice,
+} = vi.hoisted(() => {
```

Comment:
sorry just curious what's the benefit of using `hoisted()` here? why don't just `mock()`?

Reply:
thanks for the explanation 🙏